include/nbl/core/string/stringutil.h

```diff
@@ -13,6 +13,7 @@
 #include "string.h"
 #include "irrString.h" // file&class to kill
 #include "path.h" // file&class to kill
+#include <filesystem>
 
 namespace nbl
```

Comment:
most of the reason why some of the templates in this in header exist is because `stringc`didnt expose everything a C++11 `std::string` would 

another was because we didn't use `std::string_view` and used a lot of `char*` where in fact we should have been using a stringview

maybe mark some "candidates for deletion" if they are redundant